wordpress-mobile/WordPress-Android
# Feature/move me tab to new activity

Fixes #10495 

As a part of the Information Architecture Project this PR aims to move the access to the Me section, from being a menu in the main Bottom Navigation Bar to being a dedicated icon in the My Site section options menu. (credits also to @planarvoid since great part of this code changes was done during a pair programming session).

The following images show a before and after of this:

 | Before | After |
|---|---|
| ![before](https://user-images.githubusercontent.com/47797566/64516005-f4f09380-d2ed-11e9-9833-be66c9e351cc.png) | ![after](https://user-images.githubusercontent.com/47797566/64516027-0043bf00-d2ee-11e9-9d8f-00e77d0157a9.png) |

**Notes**
Some notes on this current first iteration implementation:
- As explained in #10495 to get a "balanced" Bottom Navigation Bar with 3 items (My site, Reader, Me) this PR needs to be shipped together with the PR relevant to the post creation button to be transformed in a FAB (this is currently under design). For now we removed only the Me menu item from the nav bar
- There has been some elaboration if to use the user Gravatar or the generic user icon for the added menu. We used the "ic_user_circle_white_24dp" for this first iteration. We can decide if to stay with it or change to Gravatar in a future iteration. (is this correct @osullivanchris ?)
- To feature flag this enhancement, we used a boolean flag (`ME_ACTIVITY_AVAILABLE`) in the WordPress build.gradle [here](https://github.com/wordpress-mobile/WordPress-Android/blob/bf845ad88805ce2ab9059c6b986f4b2bf6382aef/WordPress/build.gradle#L86) . The feature is disabled for vanilla flavor and enabled in the other cases.

This is the current result:

<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/47797566/64521845-07bc9580-d2f9-11e9-863e-406cdee4128a.gif">
</p>

**Manual testing:**

1. Compile and run the wasabi flavor
2. Check that the Me menu item is not in the bottom nav bar
<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/47797566/64517898-7dbcfe80-d2f1-11e9-9905-5cd15df45293.png">
</p>

3. Check that the me options menu icon
<p align="center">
  <img width="40" src="https://user-images.githubusercontent.com/47797566/64518186-03d94500-d2f2-11e9-84a4-7f56ada2288e.png">
</p>
 is only available in the My site section

4. Tap on each menu item in the bottom navbar to check that the correct screen is visualized
5. Tap on the me options menu icon to check that the Me section is visualized
6. Tap on the back arrow and check you go back to the My site section
7. Compile and run vanilla flavor
8. check that the My site Me option menu is not available
9. Check that you have all the 5 menu item in the Bottom nav bar
10. Smoke test to check that the menus activate the expected screens. 


**Update release notes:**
